src/Graph/GraphFormatter.php

```diff
@@ -0,0 +1,229 @@
+						$this->options['wordWrapLimit'] );
+				}
+			}
+
+			/**
+			 * Add nodes to the graph
+			 *
+			 * @var \SRF\Graph\GraphNode $node
+			 */
+			$this->add( "\"" . $node->getID() . "\"" );
+
+			if ( $this->options['enableGraphLink'] ) {
+
+				$nodeLinkURL = "[[" . $node->getID() . "]]";
+
+				if( $nodeLabel === '' ) {
+					$this->add( " [URL = \"$nodeLinkURL\"]" );
+				} else {
+					$this->add( " [URL = \"$nodeLinkURL\", label = \"$nodeLabel\"]" );
+				}
+			}
+			$this->add( "; ");
+		}
+
+		/**
+		 * Add edges to the graph
+		 *
+		 * @var \SRF\Graph\GraphNode $node
+		 */
+		foreach ( $nodes as $node ) {
+
+			if ( count( $node->getParentNode() ) > 0 ) {
+
+				foreach ( $node->getParentNode() as $parentNode ) {
+
+					// handle parent/child switch (parentRelation)
+					$this->add( $this->options['parentRelation'] ? " \"" . $parentNode['object']
+						. "\" -> \"" . $node->getID() . "\""
+						: " \"" . $node->getID() . "\" -> \"" . $parentNode['object'] . "\" " );
+
+					if ( $this->options['showGraphLabel'] || $this->options['showGraphColor'] ) {
+						$this->add( ' [' );
+
+						// add legend item only if missing
+						if ( array_search( $parentNode['predicate'], $this->legendItem, true ) === false ) {
+							$this->legendItem[] = $parentNode['predicate'];
+						}
+
+						// assign color
+						$color = $this->graphColors[array_search( $parentNode['predicate'], $this->legendItem, true )];
+
+						// show arrow label (graphLabel is misleading but kept for compatibility reasons)
+						if ( $this->options['showGraphLabel'] ) {
+							$this->add( "label=\"" . $parentNode['predicate'] . "\"" );
+							if ( $this->options['showGraphColor'] ) {
+								$this->add( ",fontcolor=$color," );
+							}
+						}
+
+						// colorize arrow
+						if ( $this->options['showGraphColor'] ) {
+							$this->add( "color=$color" );
+						}
+						$this->add( "]" );
+					}
+					$this->add( ";" );
+				}
+			}
+		}
+		$this->add( "}" );
+	}
+
+	/**
+	 * Creates the graph legend
+	 *
+	 * @return string Html::rawElement
+	 *
+	 */
+	public function getGraphLegend(){
+		$itemsHtml = '';
+		$colorCount = 0;
+		$arraySize = count( $this->graphColors );
+
+		if ( $this->options['showGraphLegend'] && $this->options['showGraphColor'] ) {
+			foreach ( $this->legendItem as $legendLabel ) {
+				if ( $colorCount >= $arraySize ) {
+					$colorCount = 0;
+				}
+
+				$color = $this->graphColors[$colorCount];
+				$itemsHtml .= Html::rawElement( 'div', [ 'class' => 'graphlegenditem', 'style' => "color: $color" ],
+					"$color: $legendLabel" );
+
+				$colorCount ++;
+			}
+		}
+
+		return Html::rawElement( 'div', [ 'class' => 'graphlegend' ], "$itemsHtml");
+	}
+
+	/**
+	 * Returns the word wrapped version of the provided text.
+	 *
+	 * @param string $text
+	 * @param integer $charLimit
+	 *
+	 * @return string
+	 */
+	public static function getWordWrappedText( $text, $charLimit ) {
```

Comment:
Did you make this public and static so you can test it?

Reply:
Also, you can do

```php
getWordWrappedText( string $text, int $charLimit ): string {
```

and then drop the whole docblock

Reply:
No I think that "public static" is best for this function because it is not bound to a GraphFormatter object.

Reply:
Maybe it should be its own class? (I'm not sure since I have not looked at this long enough)

Global functions are generally bad, and static functions are global functions. Then again this one is not accessing any state, and pure function leaf nodes can be global without too much issues.